src/wsproxy/lib/backend.ai-ws-appproxy.js

```diff
@@ -54,21 +54,27 @@ module.exports = (proxy = class Proxy extends ai.backend.Client {
     return hdrs;
   }
 
-  start(sessionName, app, ip, port) {
+  start(sessionName, app, ip, port, envs = {}, args = {}) {
     return new Promise(resolve => {
       this._resolve = resolve;
-      this._start(sessionName, app, ip, port);
+      this._start(sessionName, app, ip, port, envs, args);
     });
   }
 
-  _start(sessionName, app, ip, port) {
+  _start(sessionName, app, ip, port, envs, args) {
     this.ip = ip;
     this.port = port;
-    this._conn(sessionName, app);
+    this._conn(sessionName, app, envs, args);
     this.tcpServer.listen(this.port, this.ip);
   }
 
-  _conn(sessionName, app) {
+  _conn(sessionName, app, envs, args) {
+    if (typeof envs === 'object' && Object.keys(envs).length > 0) {
+      app = app + '&envs='+ JSON.stringify(envs);
```

Comment:
I think this should be url-encoded for safety. If url is encoded here, the receiver (`manager.js`) should also url-decode before parsing JSON. For example, I'm not sure the following `envs` will be delivered properly as a querystring.

```javascript
'&envs='+JSON.stringify({myenv: 'ab&cd=e'})
> "&envs={"myenv":"ab&cd=e"}"
```